docs/source/relayer.rst

```diff
@@ -1,2 +1,273 @@
+        port: 3311,
+        privateKey: 'YOUR_PRIVATE_KEY',
+        chainSpec: {
+          ensAddress: '0xe7410170f87102DF0055eB195163A03B7F2Bff4A',
+          chainId: 0
+        },
+        ensRegistrars: [
+          'poppularapp.test', 
+          'my-id.test', 
+          'my-super-domain.test'
+        ]
+
+        const knexConfig = {
+        client: 'postgresql',
+        connection: {
+          database: 'universal_login_relayer_development',
+          user:     'postgres',
+          password: 'postgres'
+        },
+        migrations: {
+          directory: path.join(__dirname, './node_modules/universal-login-relayer/migrations')
+        }
+      };
+    
+      const database = knex(knexConfig);
+      const relayer = new Relayer(config, '', database);
+      relayer.start();
+
+
+Personalised relayer
+--------------------
+
+After every operations on contract, there is emitted an event. You can add listeners to this events and transfer funds for every operation.
+
+**this.hooks.addListener('eventType', callback)**
+
+  subscribes an event.
+
+  Parameters:
+    - **eventType** : string - type of event, possible event types: ``created``, ``added`` and  ``keysAdded``
+    - **callback**
+  Returns: 
+    event listener
+  In this example, we create token granting relayer, that gives tokens to wallet contract for creation, adding key or adding keys. 
+    ::
+
+      import ethers from 'ethers';
+      import waitToBeMined from 'universal-login-contracts';
+      import Token from './build/Token';
+
+      class TokenGrantingRelayer extends Relayer {
+        constructor(config, provider = '', database) {
+          super(config, provider, database);
+          this.tokenContractAddress = config.tokenContractAddress;
+          this.tokenContract = new ethers.Contract(this.tokenContractAddress, Token.interface, this.wallet);
+          this.addHooks();
+        }
+
+        addHooks() {
+          this.hooks.addListener('created', async (transaction) => {
+            const receipt = await waitToBeMined(this.provider, transaction.hash);
+            if (receipt.status) {
+              this.tokenContract.transfer(receipt.contractAddress, ethers.utils.parseEther('100'));
+            }
+          });
+
+          this.addKeySubscription = this.hooks.addListener('added', async (transaction) => {
+            const receipt = await waitToBeMined(this.provider, transaction.hash);
+            if (receipt.status) {
+              this.tokenContract.transfer(transaction.to, ethers.utils.parseEther('5'));
+            }
+          });
+
+          this.addKeysSubscription = this.hooks.addListener('keysAdded', async (transaction) => {
+            const recepit = await waitToBeMined(this.provider, transaction.hash);
+            if (recepit.status) {
+              this.tokenContract.transfer(transaction.to, ethers.utils.parseEther('15'));
+            }
+          });
+        }
+      }
+
```

Comment:
Note: relayer will issue a new transaction after contract is deployed. Therefore ether/tokens will not appear instantly, but after a while